### PR TITLE
nixosTests.openarena: port to python

### PR DESCRIPTION
**Pull request by mmilata**
###### Motivation for this change

#72828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tomfitzhenry

### CODE REVIEW

File: nixos/tests/openarena.nix [hunk 1]
```diff
@@ -23,14 +23,19 @@ import ./make-test.nix ({ pkgs, ...} : {
...
       # The log line containing 'resolve address' is last and only message that occurs after
       # the server starts accepting clients.
-      $machine->waitUntilSucceeds("journalctl -u openarena.service | grep 'resolve address: dpmaster.deathmask.net'");
+      machine.wait_until_succeeds(
+          "journalctl -u openarena.service | grep 'resolve address: dpmaster.deathmask.net'"
+      )
 
       # Check it's possible to join the server.
-      $machine->succeed("echo -n -e '\\xff\\xff\\xff\\xffgetchallenge' | socat - UDP4-DATAGRAM:127.0.0.1:27960 | grep -q challengeResponse");
+      # Can't use substring match instead of grep because the output is not utf-8
```
Comment (tomfitzhenry) [inline nixos/tests/openarena.nix]: Huh! I recall trying to use substring matcihng and it failing, and I didn't look into why. Good to know!

**Review by tomfitzhenry — APPROVED** [review]
LGTM. I'm not sure how to trigger the test to ensure it still passes, but the code LGTM.

**Comment by mmilata** [comment]
@tomfitzhenry Either `nix-build nixos/tests/openarena.nix` locally or comment `@GrahamcOfBorg test openarena` if you have the rights.

**Review by tfc — APPROVED** [review]

**Comment by flokli** [comment]
Verified test still succeed.

**Pull request merged**